Pigmice2733/peregrine-frontend
# Switch dev server + prod build to vite!

Wheeee!

![image](https://github.com/Pigmice2733/peregrine-frontend/assets/13206945/4eef4e32-e17f-49c1-892e-3155b89082ff)

Finally got this working! The first time I tried to switch this project over to vite was in 2019 when vite was first published. About every year since then, I've come back to this trying to get it to work but there have always been roadblocks (ehem, Linaria, Jest, ESM support...). Last year I got pretty close but now I had some time to finally get it across the finish line!

Before, we were using Webpack 4 (very old) for the dev server and Rollup for the production build. Now, we are using Vite for both (well, Vite uses Rollup under the hood for the build).

Sorry, this is a big PR, but it had to be. I didn't see any way to incrementally migrate the build system over.

Testing:
- make sure the local dev server works:
  - Make sure you have node 18+ installed
  - Run `npm install`
  - Run `npm run dev`
  - Open localhost:2733 in your browser, click through the app, make sure things are working
  - Make local changes to files and make sure they are injected into the localhost version of the app
- Make sure the production build works:
  - Make sure you have node 18+ installed
  - Run `npm install`
  - Run `npm run build`
  - Run `npm run preview` to startup a preview server of the production-built app
  - Open localhost:2733 in your browser, click through the app, make sure things are working
